zapcore/write_syncer_test.go

```diff
@@ -56,6 +56,16 @@ func TestAddSyncWriteSyncer(t *testing.T) {
 	assert.Error(t, ws.Sync(), "Expected to propagate errors from concrete type's Sync method.")
 }
 
+func TestLockedWriteSyncerInterface(t *testing.T) {
+	buf := &bytes.Buffer{}
+	ws := AddSync(buf)
+	var lws WriteSyncer = &lockedWriteSyncer{ws: ws}
```

Comment:
Yes, half part. It also makes sure the internal `lockedWriteSyncer` is okay.
A separately implementation would be almost same as our default one, so I don't think it necessary.
Would some rewords help? Any thoughts?